### DISCUSSION TITLE
Defaulting the flashlight to on.

### DISCUSSION THREAD

**Discussion by ChapHop** [capawesome-team/capacitor-mlkit#191] [Q&A] [answered]
This is the code I've attempted to use to default the flashlight to on while scanning; any idea what I'm doing wrong?

```
async startScanner() {
  await BarcodeScanner.enableTorch();
  const { barcodes } = await BarcodeScanner.scan({
      formats: [BarcodeFormat.Ean13,BarcodeFormat.Code39,BarcodeFormat.Code93,BarcodeFormat.Code128]
  });
  return barcodes;
}

**Comment by robingenz** [comment, marked as answer]
You must enable the flashlight **after** starting the scan. Alternatively, you can test whether it works with the new [Torch](https://capawesome.io/plugins/torch/) plugin.
